src/audio/codec_adapter/codec_adapter.c

```diff
@@ -324,18 +324,75 @@ static int codec_adapter_copy(struct comp_dev *dev)
 	return ret;
 }
 
+static int ca_set_binary_data(struct comp_dev *dev,
+			      struct sof_ipc_ctrl_data *cdata)
+{
+	int ret;
+
+	comp_info(dev, "ca_set_binary_data() start, data type %d",
+		  cdata->data->type);
+
+	switch (cdata->data->type) {
+	case CODEC_CFG_SETUP:
+	case CODEC_CFG_RUNTIME:
+		//TODO:
+		comp_warn(dev, "ca_set_binary_data() set_data not implemented yet.");
+		ret = -EIO;
+		break;
+	default:
+		comp_err(dev, "ca_set_binary_data() error: unknown binary data type");
+		ret = -EINVAL;
+		break;
+	}
+
+	return ret;
+}
+
+static int codec_adapter_ctrl_set_data(struct comp_dev *dev,
+				       struct sof_ipc_ctrl_data *cdata)
+{
```

Comment:
All static functions in this file should have the same prefix. Either ``codec_adapter`` or ``ca`` but not both. I would prefer ``ca`` as it is shorter, but any works for me.